ethcore/src/snapshot/service.rs

```diff
@@ -656,18 +689,65 @@ impl Service {
 		match self.feed_chunk(hash, chunk, false) {
 			Ok(()) => (),
 			Err(e) => {
-				warn!("Encountered error during block restoration: {}", e);
+				match e.kind() {
+					::error::ErrorKind::Snapshot(SnapshotError::RestorationAborted) => {
+						debug!(target: "snapshot", "Restoration aborted.");
+					},
+					_ => {
+						warn!("Encountered error during block restoration: {}", e);
+						let _ = fs::remove_dir_all(self.restoration_dir());
+					},
+				}
 				*self.restoration.lock() = None;
 				*self.status.lock() = RestorationStatus::Failed;
 				let _ = fs::remove_dir_all(self.restoration_dir());
 			}
 		}
 	}
+
+	/// Get the completed Snapshot Manifest (if any), read from FS
+	fn completed_manifest(&self) -> Option<ManifestData> {
+		self.reader.read().as_ref().map(|r| r.manifest().clone())
+	}
+
+	/// Get the currently partially restored Snapshot Manifest (if any)
+	fn partial_manifest(&self) -> Option<ManifestData> {
```

Comment:
`partial_manifest` name is slightly misleading, since it is a snapshot that is partial, not a manifest, if IIUC